src/server/zset_family.cc

```diff
@@ -1991,28 +2013,14 @@ void ZSetFamily::ZInterStore(CmdArgList args, ConnectionContext* cntx) {
   cntx->transaction->Schedule();
   cntx->transaction->Execute(std::move(cb), false);
 
-  ScoredMap result;
-  for (auto& op_res : maps) {
-    if (op_res.status() == OpStatus::SKIPPED)
-      continue;
-
-    if (!op_res)
-      return (*cntx)->SendError(op_res.status());
-
-    if (result.empty()) {
-      result.swap(op_res.value());
-    } else {
-      InterScoredMap(&result, &op_res.value(), op_args.agg_type);
-    }
-
-    if (result.empty())
-      break;
```

Comment:
I see that previous code did not have the bug as it checked afterward if things are empty. We could use this logic, it's better :)

Reply:
fix: https://github.com/dragonflydb/dragonfly/pull/2164/commits/00842a80fcc12b92febd0c2330b4a08882c56722#diff-294d1289b1741ae808c06f38d260077a53dcba7762c845f9677a068568821fc9R1089-R1091